explainer.md

```diff
@@ -112,7 +112,36 @@ interface LaunchEvent : ExtendableEvent {
 ```
 Notes:
 * The `Request` that led to the navigation is included. In addition to the URL, this allows sites to inspect things like POST data (e.g. for a Web Share Target) that led to the invocation of the app. 
-## Design Questions/Concerns
+
+## Design Questions/Details
+
+### Restricting to installed apps
+
+It is recommended that user agents only fire launch events for installed web apps. There are 2 reasons for this:
+ 1. It is difficult to attribute bad behavior to misbehaving websites if they aren't installed (see the section below).
+ 2. It could be confusing if the behavior of clicking a link changes just because a user has visited a site that registered itself as a `launch` event handler. 
+
+Allowing launch events to be handled on the drive-by web could be explored in the future.
+
+### Handling Redirects
+
+We should allow launch events to be intercepted when a navigation triggers a redirect. For example imagine that https://www.example.com registers a `launch` event handler. The user then clicks a link to https://example.com. No handler is registered, so the navigation is processed. The navigation redirects to https://www.example.com. At this point, before the request is actually made, a launch event should be fired on the SW for https://www.example.com.
+
+If we don't do this, then many link clicks won't be correctly intercepted.
+
+User agents need to decide what the underlying page will do if a redirect is captured in separate window. The options are:
```

Comment:
I think we should make a recommendation here. Some of these suggestions would break sites (stopping the navigation, returning to the page before the navigation) if browsers implement them differently.

Reply:
This seems super relevant though. I think messenger.com throws you through a tracking redirect for all links :/

Reply:
Agreed - adjusted the language